app/controllers/api/v4/application_controller.rb

```diff
@@ -55,7 +55,11 @@ def set_category
   end
 
   def set_game
-    @game = Game.joins(:srdc).find_by!(speedrun_dot_com_games: {shortname: params[:game]})
+    if params[:game_value] == 'id'
+      @game = Game.find(params[:game])
+    else
+      @game = Game.joins(:srdc).find_by!(speedrun_dot_com_games: {shortname: params[:game]})
+    end
```

Comment:
this is safe and efficient, but i think the interface can be simpler. have you considered something like

1. let `params[:game]` be an ID or a shortname without telling us which, we take the extra 1-query hit and keep the understanding of how to use the API dead simple, and accept the possibility of collisions. or
2. merge the lookup-by-ID function into the `/v4/games?search=:game` endpoint, where if you supply just an ID we return an array containing only that game.

Reply:
I was really trying to avoid the collision angle because as a developer using it I would not want to pass in what I thought to be a game ID and then get back some other result.  Plus the fact that it might be a valid game ID at first but then a new shortname gets added with the same ID which could then change the result.

I'm not sure I understand what you're proposing for option 2.  Do you mean the flag option to the search endpoint?  Or do you mean accept the collisions there since we'd be returning multiple results so it wouldn't be a problem?  Or am I totally off the mark with both of those.

Reply:
For posterity, we're going with having the game search check and see if an ID exists with that exact string, and if it does insert the matching record included as part of the search (but include the search results as well to preserve original behavior).